debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java

```diff
@@ -112,14 +113,24 @@ public BinlogReader(String name, MySqlTaskContext context) {
         EventDeserializer eventDeserializer = new EventDeserializer() {
             @Override
             public Event nextEvent(ByteArrayInputStream inputStream) throws IOException {
-                // Delegate to the superclass ...
-                Event event = super.nextEvent(inputStream);
-                // We have to record the most recent TableMapEventData for each table number for our custom deserializers ...
-                if (event.getHeader().getEventType() == EventType.TABLE_MAP) {
-                    TableMapEventData tableMapEvent = event.getData();
-                    tableMapEventByTableId.put(tableMapEvent.getTableId(), tableMapEvent);
+                try {
+                    // Delegate to the superclass ...
+                    Event event = super.nextEvent(inputStream);
+
+                    // We have to record the most recent TableMapEventData for each table number for our custom deserializers ...
+                    if (event.getHeader().getEventType() == EventType.TABLE_MAP) {
+                        TableMapEventData tableMapEvent = event.getData();
+                        tableMapEventByTableId.put(tableMapEvent.getTableId(), tableMapEvent);
+                    }
+                    return event;
+                }
+                // DBZ-217 In case an event couldn't be read we create a pseudo-event for the sake of logging
+                catch(EventDataDeserializationException edde) {
+                    EventHeaderV4 header = new EventHeaderV4();
+                    header.setEventType(EventType.INCIDENT);
+                    EventData data = new EventDataDeserializationExceptionData(edde);
+                    return new Event(header, data);
```

Comment:
IIRC, we can get this exception for other cases, like unknown MySQL event types. That would be rare, but I'm wondering if we should constrain this as much as possible. What are the potential cases other than the corrupt binlog even where this might be thrown?

Reply:
It's an option now, with the default being the current behavior, i.e. stopping the connector, but logging the context required to analyze the event in the binlog. I think the DLQ approach is nice, but something we can handle separately and add it as further value of that new option.